Angular InstantSearch/refresh/src/app/refresh.component.ts

```diff
@@ -1,39 +1,75 @@
-import { Component, Inject, forwardRef, OnInit } from "@angular/core";
-import { BaseWidget, NgAisInstantSearch } from "angular-instantsearch";
+import { Component, Inject, forwardRef, Optional, Input } from '@angular/core';
+import {
+  TypedBaseWidget,
+  NgAisInstantSearch,
+  NgAisIndex,
+} from 'angular-instantsearch';
+import { Connector } from 'instantsearch.js/es/types';
 
-const connectRefresh = (renderFn, unmountFn) => (widgetParams = {}) => ({
-  init() {
-    renderFn({ refresh() {} }, true);
-  },
-  render({ instantSearchInstance }) {
-    const refresh = instantSearchInstance.refresh.bind(instantSearchInstance);
-    renderFn({ refresh }, false);
-  },
-  dispose() {
-    unmountFn();
-  }
-});
+type RefreshWidgetDescription = {
+  $$type: 'custom.referesh',
+  renderState: {
+    refresh(): void;
+  };
+};
+type RefreshConnectorParams = {};
+
+type RefreshConnector = Connector<
+  RefreshWidgetDescription,
+  RefreshConnectorParams
+>;
+
+const connectRefresh: RefreshConnector =
+  (renderFn, unmountFn) => (widgetParams) => ({
+    $$type: 'custom.referesh',
+    init: function ({ instantSearchInstance }) {
+      renderFn(
+        {
+          refresh() {},
+          widgetParams,
+          instantSearchInstance,
+        },
+        true
+      );
+    },
+    render({ instantSearchInstance }) {
+      const refresh = instantSearchInstance.refresh.bind(instantSearchInstance);
+      renderFn({ refresh, widgetParams, instantSearchInstance }, false);
```

Comment:
In the `init` method we use multiple lines for `renderFn` so we can do the same here for consistency.

Reply:
fixed here: 
edf77ef